### DISCUSSION TITLE
Stop bundler after bundling

### DISCUSSION THREAD

**Discussion by nickramos94** [parcel-bundler/parcel#4664] [General] [answered]
Hi,
I'm testing Parceljs in my Node app and everything works _almost_ just fine: the bundler successfully bundles the site I ask him to bundle but after that the application never goes on with its execution and the bundler is never closed or terminated.

```
 /home/nicola/Documenti/parcel/public/index.html
⠸ Building css-loader.js...Successfully uploaded 'index.html'!
⠼ Building hmr-runtime.js...Successfully uploaded 'style.e308ff8e.js'!
⠴ Building hmr-runtime.js...Successfully uploaded 'style.e308ff8e.css.map'!
Successfully uploaded 'style.e308ff8e.js.map'!
Successfully uploaded 'style.e308ff8e.css'!
✨  Built in 979ms.
```

This is the code with which I'm doing the testing

```
const fs = require('fs');
const path = require('path');
const Bundler = require('parcel-bundler');

const file = path.join(__dirname, './public/index.html');

const options = 
{
    outDir: './dist',
    outFile: 'index.html',
    publicUrl: '/',
    target: 'browser',
};

const bundle = async () =>
{
    const done = await bundler.bundle();
};


const bundler = new Bundler(file, options);
bundle();


bundler.on('buildEnd', () =>
{
    console.log('Success');
    return;
});
```

On the API section of Parcel's documentation I didn't find any function of the bundler that lets me stop it, can somebody help me? 

Thanks

**Comment by 101arrowz** [comment, marked as answer]
If you're not running with `NODE_ENV` set to `'production'`, Parcel defaults to watching the files for changes and rebuilding on change, so it won't ever terminate on its own.

To fix this, just add `watch: false` in options or (preferably, since it seems you are building in production anyway) set `NODE_ENV` to `'production'`.
